src/command_modules/azure-cli-network/azure/cli/command_modules/network/commands.py

```diff
@@ -377,7 +377,7 @@
 cli_command(__name__, 'network dns zone list', 'azure.cli.command_modules.network.custom#list_dns_zones')
 cli_generic_update_command(__name__, 'network dns zone update',
                            'azure.mgmt.dns.operations.zones_operations#ZonesOperations.get',
-                           'azure.mgmt.dns.operations.zones_operations#ZonesOperations.create_or_update',
```

Comment:
After our discussion, you should use the create_or_update from the SDK as the setter. You don't need a custom setter.